platform/admin/modules/admin-docs-ui/src/main/java/org/codice/ui/admin/docs/DocsServlet.java

```diff
@@ -0,0 +1,226 @@
+
+    private static final String FILTER_MAPPING_PARAM = "filterMappingUrlPattern";
+
+    private List<SparkApplication> sparkApplication =
+            Collections.synchronizedList(new ArrayList<>());
+
+    private static MatcherFilter matcherFilter;
+
+    public DocsServlet(List<SparkApplication> application) {
+        sparkApplication = application;
+    }
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+        ServletFlag.runFromServlet();
+
+        sparkApplication.stream()
+                .sequential()
+                .forEach(SparkApplication::init);
+
+        matcherFilter = new MatcherFilter(ServletRoutes.get(),
+                StaticFilesConfiguration.servletInstance,
+                true,
+                false);
+    }
+
+    @Override
+    public void destroy() {
+        if (sparkApplication != null) {
+            sparkApplication.stream()
+                    .filter(Objects::nonNull)
+                    .forEach(SparkApplication::destroy);
+        }
+    }
+
+    private String getFilterPath() throws ServletException {
+        Optional<String> mapping = this.getServletContext()
+                .getServletRegistration(this.getServletName())
+                .getMappings()
+                .stream()
+                .findFirst();
+        if (mapping.isPresent()) {
+            String result = mapping.get();
+            if (result == null || result.equals(SLASH_WILDCARD)) {
+                return "";
+            } else if (!result.startsWith(SLASH) || !result.endsWith(SLASH_WILDCARD)) {
+                throw new RuntimeException("The " + FILTER_MAPPING_PARAM
+                        + " must start with \"/\" and end with \"/*\". It's: " + result);
+            }
+            return result.substring(1, result.length() - 1);
+        } else {
+            throw new ServletException("Unable to read servlet mapping");
+        }
+    }
+
+    @Override
+    protected void service(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+
+        final String requestPath = getFilterPath();
+        final String relativePath = getRelativePath(req, requestPath);
+
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug(relativePath);
+        }
+
+        HttpServletRequestWrapper requestWrapper = new HttpServletRequestWrapper(req) {
+
+            @Override
+            public String getServletPath() {
+                return SLASH;
+            }
+
+            @Override
+            public String getPathInfo() {
+                return relativePath;
+            }
+
+            @Override
+            public String getRequestURI() {
+                return relativePath;
+            }
+        };
+        matcherFilter.doFilter(requestWrapper, resp, null);
+    }
+
+    static String getRelativePath(HttpServletRequest request, String filterPath) {
+        String path = request.getRequestURI();
```

Comment:
We tested this by using  paths like `admin/documentation/..` and we were unable to break out of the documentation directory